source/taskmanager-client/src/main/java/nl/aerius/taskmanager/client/mq/RabbitMQWorkerMonitor.java

```diff
@@ -104,51 +108,79 @@ public void removeObserver(final RabbitMQWorkerObserver observer) {
    * @throws IOException
    */
   public void start() throws IOException {
-    final Connection c = factory.getConnection();
-    channel = c.createChannel();
-    channel.exchangeDeclare(AERIUS_EVENT_EXCHANGE, EXCHANGE_TYPE);
-    final String queue = channel.queueDeclare().getQueue();
-    channel.queueBind(queue, AERIUS_EVENT_EXCHANGE, "");
-
-    consumer = new DefaultConsumer(channel) {
-      @Override
-      public void handleDelivery(final String consumerTag, final Envelope envelope, final AMQP.BasicProperties properties, final byte[] body)
-          throws IOException {
-        RabbitMQWorkerMonitor.this.handleDelivery(properties);
+    tryStartingConsuming.set(true);
+    while (!isShutdown) {
+      try {
+        stopAndStartConsumer();
+        LOG.debug("Successfully (re)started consumer RabbitMQWorkerMonitor");
+        if (consumer.getChannel().isOpen()) {
+          tryStartingConsuming.set(false);
+          break;
+        }
+      } catch (final ShutdownSignalException | IOException e1) {
+        LOG.warn("(Re)starting consumer RabbitMQWorkerMonitor failed, retrying", e1);
+      }
+    }
+  }
+
+  private void stopAndStartConsumer() throws IOException {
+    synchronized (this) {
+      if (consumer != null) {
+        try {
+          if (consumer.getChannel().isOpen()) {
+            consumer.getChannel().basicCancel(queueName);
+          }
+        } catch (final AlreadyClosedException | IOException e) {
+          LOG.debug("Exception while stopping consuming, ignoring.", e);
+        }
       }
+      final Connection c = factory.getConnection();
+      channel = c.createChannel();
+      channel.exchangeDeclare(AERIUS_EVENT_EXCHANGE, EXCHANGE_TYPE);
+      queueName = channel.queueDeclare().getQueue();
+      channel.queueBind(queueName, AERIUS_EVENT_EXCHANGE, "");
+      consumer = new DefaultConsumer(channel) {
+        @Override
+        public void handleDelivery(final String consumerTag, final Envelope envelope, final AMQP.BasicProperties properties, final byte[] body)
+            throws IOException {
+          RabbitMQWorkerMonitor.this.handleDelivery(properties);
+        }
 
-      @Override
-      public void handleShutdownSignal(final String consumerTag, final ShutdownSignalException sig) {
-        if (sig.isInitiatedByApplication()) {
-          LOG.info("Worker event monitor {} was shut down by the application.", consumerTag);
```

Comment:
Maybe set `shutdown` here to `true`? In case this event it thrown and the application is trying to (re)start?